java/system-test/system-test-dsl/src/main/java/sleeper/systemtest/dsl/instance/DeployedSleeperInstance.java

```diff
@@ -54,17 +55,21 @@ private DeployedSleeperInstance(
     }
 
     public static DeployedSleeperInstance loadOrDeployIfNeeded(
-            String instanceId, DeployInstanceConfiguration configuration,
+            String instanceId, SystemTestInstanceConfiguration configuration,
             SystemTestParameters parameters, DeployedSystemTestResources systemTest,
             SleeperInstanceDriver driver, AssumeAdminRoleDriver assumeRoleDriver) {
-
-        boolean newInstance = driver.deployInstanceIfNotPresent(instanceId, configuration);
+        DeployInstanceConfiguration deployConfig = configuration.buildDeployConfig(parameters, systemTest);
+        boolean newInstance = driver.deployInstanceIfNotPresent(instanceId, deployConfig);
 
         InstanceProperties instanceProperties = new InstanceProperties();
         driver.loadInstanceProperties(instanceProperties, instanceId);
+        if (newInstance && configuration.shouldEnableTransactionLogSnapshots()) {
+            instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_CREATION_RULE, "true");
+            driver.saveInstanceProperties(instanceProperties);
+        }
```

Comment:
This should apply the setting from SystemTestInstanceConfiguration by either enabling or disabling the rule. It can use the new driver for this. The instance property is the name of the rule, as set by the CDK.

It should apply whether the instance is new or is being reused from a previous execution. Note that this method is only called the first time the instance is used during this execution of JUnit.